### PR TITLE
perf: reduce `cmp.visible()` timeout from 1000ms to 42ms

### PR DESCRIPTION
**Pull request by llllvvuu**
fixes: #288

### CODE REVIEW

**Comment by ray-x** [comment]
1000ms is bad if LSP is slow. Is there a reason to choose `42ms`

**Comment by llllvvuu** [comment]
I did some testing until I found that the typing felt snappy. e.g. 100ms was still a bit too noticeable.

FWIW what would happen in theory if we don't wait too long, is that the panda text gets covered for a split second - which doesn't seem worth waiting a long time for. i.e. it seems better to just show the panda text immediately rather than blocking the editor (or the panda text can be shown async, but that's a larger effort)

**Review by ray-x — APPROVED** [review]
Thanks

File: lua/lsp_signature/helper.lua [hunk 1]
```diff
@@ -697,7 +697,10 @@ end
 helper.completion_visible = function()
   local hascmp, cmp = pcall(require, 'cmp')
   if hascmp then
-    return cmp.visible()
+    -- reduce timeout from cmp's hardcoded 1000ms:
+    -- https://github.com/ray-x/lsp_signature.nvim/issues/288
+    cmp.core.filter:sync(42)
```
Comment (abhinavgaba) [inline lua/lsp_signature/helper.lua:702]: Looks like blink.compat, which is used to support cmp compatible sources, has `cmp.core.view:visible`, but not `cmp.core.filter`, which is leading to a crash in this line.

I created an issue ticket there to check if it's feasible to add `filter:sync` in blink.compat here: https://github.com/Saghen/blink.compat/issues/25

**Pull request merged**